components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.email/src/main/java/org/wso2/carbon/event/output/adapter/email/EmailEventAdapter.java

```diff
@@ -129,61 +251,64 @@ public void destroy() {
+            }
+
+
+            //Creating MIME object using initiated session.
+
+            MimeMessage message = new MimeMessage(session);
+
+            //Setting up the Email attributes and Email payload.
+
             try {
-                ConfigurationContext configContext = EmailEventAdapterServiceValueHolder
-                        .getConfigurationContextService().getClientConfigContext();
-
-                //Set configuration service client if available, else create new service client
-                if (configContext != null) {
-                    serviceClient = new ServiceClient(configContext, null);
-                } else {
-                    serviceClient = new ServiceClient();
+                message.setFrom(smtpFromAddress);
+                message.addRecipient(Message.RecipientType.TO,
+                        new InternetAddress(to));
+
+                message.setSubject(subject);
+                message.setSentDate(new Date());
+                message.setContent(body, type);
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Meta data of the email configured successfully");
                 }
-                Options options = new Options();
-                options.setProperty(Constants.Configuration.ENABLE_REST, Constants.VALUE_TRUE);
-                options.setProperty(MessageContext.TRANSPORT_HEADERS, headerMap);
-                options.setProperty(MailConstants.TRANSPORT_MAIL_FORMAT,
-                        MailConstants.TRANSPORT_FORMAT_TEXT);
-                options.setTo(new EndpointReference(EmailEventAdapterConstants.EMAIL_URI_SCHEME + to));
-
-                serviceClient.setOptions(options);
-                serviceClient.fireAndForget(payload);
-                log.debug("Sending confirmation mail to " + to);
-            } catch (AxisFault e) {
-                String msg = "Error in delivering the message, " +
-                        "subject: " + subject + ", to: " + to + ".";
-                log.error(msg);
-            } catch (Throwable t) {
-                String msg = "Error in delivering the message, " +
-                        "subject: " + subject + ", to: " + to + ".";
-                log.error(msg);
-                log.error(t);
-            } finally {
-                if (serviceClient != null) {
-                    try {
-                        serviceClient.cleanup();
-                    } catch (AxisFault axisFault) {
-                        log.error("Error while cleaning-up service client resources ", axisFault);
-                    }
+
+                Transport.send(message);
+
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Mail sent to the EmailID" + " " + to + " " + "Successfully");
                 }
+
+            } catch (MessagingException e) {
+                log.error("Message format error in sending the Email from : " +
```

Comment:
Can we add some context information here. It would have been better if we can log what the format is.